client/src/app/views/form.coffee

```diff
@@ -0,0 +1,37 @@
+$ = require('jquery')
+Marionette = require('backbone.marionette')
+template = require('../templates/form.jade')
+errorMessage = require('../templates/error.jade')
+SiteModel = require('../models/site')
+
+module.exports = Marionette.ItemView.extend(
+  model: new SiteModel()
+  template: template
+  
+  events:
+    'submit form' : 'enterSite'
+  
+  ui:
+    url: '#url'
+    email: '#email'
+  
+  enterSite: (e) ->
+    e.preventDefault()
+    unless @validateForm() then return
+    url = @ui.url.val().trim()
+    email = @ui.email.val().trim()
+    @ui.url.val('')
+    @ui.email.val('')
+    @model.save(url: url,
+      success: ->
+        location.hash = "sites"
+    )
+
+  validateForm: ->
+    $('.error').remove()
+    @ui.url.css('border', '')
+    if @ui.url.val().trim().length > 0 then return yes
+    @ui.url.css('border', '2px solid #f00')
+    @$el.append(errorMessage)
+    return no
```

Comment:
This type of css styling should just be done with classes and not manual inline rules.  For example we should just be adding or removing a class like `.error` from fields with errors instead of manually setting `border` properties inline here.

Also, use the booleans `true` and `false` instead of `yes` and `no`.

Why do we remove the `.error` DOM node only to put it right back in via `@$el.append`?  It's the same error message every time right?  We dont need to keep removing it just to reshow the same thing.  And if we do want to remove it, lets scope that lookup also to `@$('.error').remove()`.   I think we can do the validation and then if invalid, show the error and add the error class, and if we are valid, then remove the error DOM node and remove the error classes.